Allow different isolated confluence containers

Vidda777 opened the pull request:
Allow different isolated confluence containers by two changes:
- using COMPOSE_PROJECT_NAME variables that will allow isolated folders where data is saved per project/containers
- defining container names by CONFLUENCE_VERSION variable


See more here: https://docs.docker.com/compose/#multiple-isolated-environments-on-a-single-host

P.D: write me on slack :P

Commit:
Modify docker compose container names to allow different container for different confluences :-). Added a default COMPOSE_PROJECT_NAME that will allow isolated folders where data is saved per project

Vidda777 commented on .env:
```diff
@@ -4,3 +4,4 @@ CONFLUENCE_VERSION=6.0.7
 POSTGRESQL_VERSION=9.6
 JAVA_VERSION=zulu@1.8.232
 DATABASE=postgres
+COMPOSE_PROJECT_NAME=default
```
Maybe this is not required as if you don't set any name, a default internal one is always used. Although including this env var could help noticing you can use it. 🤷

Vidda777 commented on docker-compose.yml:
```diff
@@ -19,13 +19,13 @@ services:
   postgres:
     image: postgres:${POSTGRESQL_VERSION}
-    container_name: postgres
+    container_name: postgres_${CONFLUENCE_VERSION}
```
I used ${CONFLUENCE_VERSION} for all containers name suffix, so all are vinculated to the specific confluence version.

aruizca commented:
Not sure how, but I would like this approach to be optional rather than the out of the box behaviour. This approach can potentially take a huge amount of disk space.

Vidda777 commented:
> 

If you don't use the COMPOSE_PROJECT_NAME, the behaviour would be the same than previously. Only one folder would be used for store data and only one container per type would be used.

Despite using CONFLUENCE_VERSION as a suffix, during my first test not using COMPOSE_PROJECT_NAME, it didn't matter the container names were previously created as `_7.15.0` when I executed again the run script with different version (ie `6.15.10`) I saw in traces that the container name was being updated :-o but keeping its previous data.

I can do a couple of more test just to make it sure. But... I think we could remove the COMPOSE_PROJECT_NAME from `.env` file and just let people used it if they want. In the README file is documented how to achieve different isolated containers using that variable.


🤔

Commit:
Fix puppeteer initialization docker service

Vidda777 commented:
Regarding the optional approach and also simplification of the script. Take a look to this changes on this branch: https://github.com/Vidda777/docker-confluence-for-testing/compare/master...Vidda777:feature/simplify_isolated_usage?expand=1

- I removed COMPOSE_PROJECT_NAME for env.
- The run scripts will use the CONFLUENCE_VERSION as project name, **ONLY** if a new parameter, ISOLATED was set. Otherwise, old behaviour is performed.
- I added a stop script that also simplifies how to stop containers, no matter if under default project or a specific one.

Commit:
Merge remote-tracking branch 'aruizca/master'

Vidda777 commented:
Close the PR as obsolete
